ipatests/pytest_ipa/integration/firewall.py

```diff
@@ -0,0 +1,137 @@
+# Authors:
+#   Thomas Woerner <twoener@redhat.com>
+#
+# Copyright (C) 2018  Red Hat
+# see file 'COPYING' for use and warranty information
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+"""Firewall class for integration testing using firewalld"""
+
+from __future__ import absolute_import
+
+from ipapython import ipautil
+
+
+class Firewall:
+    def __init__(self, host):
+        """Initialize with host where firewall changes should be applied
+        Unmasks, enables and starts firewalld."""
+        self.host = host
+        # Unmask firewalld service
+        self.host.run_command(["systemctl", "unmask", "firewalld"])
+        # Enable firewalld service
+        self.host.run_command(["systemctl", "enable", "firewalld"])
+        # Start firewalld service
+        self.host.run_command(["systemctl", "start", "firewalld"])
+
+    def _rp_action(self, args):
+        """Run-time and permanant firewall action"""
+        cmd = ["firewall-cmd"]
+        cmd.extend(args)
+
+        # Run-time part
+        result = self.host.run_command(cmd, raiseonerror=False)
+        if result.returncode not in [0, 11, 12]:
+            # Ignore firewalld error codes:
+            #   11 is ALREADY_ENABLED
+            #   12 is NOT_ENABLED
+            raise ipautil.CalledProcessError(result.returncode, cmd,
+                                             result.stderr_text)
+
+        # Permanent part
+        result = self.host.run_command(cmd + ["--permament"],
+                                       raiseonerror=False)
+        if result.returncode not in [0, 11, 12]:
+            # Ignore firewalld error codes:
+            #   11 is ALREADY_ENABLED
+            #   12 is NOT_ENABLED
+            raise ipautil.CalledProcessError(result.returncode, cmd,
```

Comment:
It's ``CalledProcessError(returncode, cmd, output, stderr)``, but you are calling it with ``returncode, cmd, stderr``.